packages/content-conformance/src/__tests__/content-file.test.ts

```diff
@@ -78,5 +78,52 @@ field: |
       `)
       expect(file.frontmatter()).toMatchInlineSnapshot(`{}`)
     })
+
+    it('should attach a message for invalid frontmatter', () => {
+      const file = new ContentFile(`---
+description: |
+this is an invalid multi-line description.
+---
+
+# Heading 1`)
+      expect(file.value).toMatchInlineSnapshot(`
+        "---
+        description: |
+        this is an invalid multi-line description.
+        ---
+
+        # Heading 1"
+      `)
+      expect(file.frontmatter()).toMatchInlineSnapshot(`{}`)
+      expect(file.messages).toMatchInlineSnapshot(`
+        [
+          [1:1: Error parsing frontmatter: YAMLParseError: Implicit map keys need to be followed by map values at line 2, column 1:
+
+        description: |
+        this is an invalid multi-line description.
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        ],
+        ]
+      `)
+    })
+
+    it('should attach a message for empty fences', () => {
+      const file = new ContentFile(`---
+---
+
+# Heading 1`)
+      expect(file.value).toMatchInlineSnapshot(`
+        "---
+        ---
+
+        # Heading 1"
+      `)
+      expect(file.frontmatter()).toMatchInlineSnapshot(`{}`)
+      expect(file.messages).toMatchInlineSnapshot(`
+        [
+          [1:1: Error parsing frontmatter: TypeError: Cannot read properties of undefined (reading 'length')],
+        ]
+      `)
+    })
```

Comment:
💭 For now, this is the best error we get, given the implementation of `vfile-matter`
- https://github.com/vfile/vfile-matter/blob/main/index.js#L47